Lib/test/test_pathlib/test_pathlib_abc.py

```diff
@@ -1822,6 +1822,170 @@ def test_copy_empty(self):
         self.assertTrue(target.exists())
         self.assertEqual(target.read_bytes(), b'')
 
+    def test_copytree_simple(self):
+        base = self.cls(self.base)
+        source = base / 'dirC'
+        target = base / 'copyC'
+        source.copytree(target)
+        self.assertTrue(target.is_dir())
+        self.assertTrue(target.joinpath('dirD').is_dir())
+        self.assertTrue(target.joinpath('dirD', 'fileD').is_file())
+        self.assertEqual(target.joinpath('dirD', 'fileD').read_text(),
+                         "this is file D\n")
+        self.assertTrue(target.joinpath('fileC').is_file())
+        self.assertTrue(target.joinpath('fileC').read_text(),
+                        "this is file C\n")
+
+    def test_copytree_complex(self):
+        def ordered_walk(path):
+            for dirpath, dirnames, filenames in path.walk(follow_symlinks=True):
+                dirnames.sort()
+                filenames.sort()
+                yield dirpath, dirnames, filenames
+        source = self.cls(self.base)
+        target = self.tempdir() / 'target'
+
+        # Special cases tested elsehwere
+        source.joinpath('dirE').rmdir()
+        if self.can_symlink:
+            source.joinpath('brokenLink').unlink()
+            source.joinpath('brokenLinkLoop').unlink()
+            source.joinpath('dirB', 'linkD').unlink()
```

Comment:
We're only removing them if `can_symlink` is *true*. These are specifically broken or infinitely looping links and we'd expect `copytree(follow_symlinks=True)` to fail.

... but it's weird and I should probably create a directory structure specifically for the `copytree()` tests, rather than reusing the one from `setUp()`.